packages/utils/src/PromiseUtils.ts

```diff
@@ -1,16 +1,20 @@
 import CanceledPromiseError from './CanceledPromiseError';
 import TimeoutError from './TimeoutError';
 
+interface CancelablePromise<T> extends Promise<T> {
+  cancel: () => void;
+}
+
 class PromiseUtils {
-  static isCancelablePromise(promise) {
+  static isCancelablePromise(promise: CancelablePromise<unknown>): boolean {
```

Comment:
Make this a type guard that returns a type predicate. The idea is that this function is called with some unknown value and we want to know if it's a cancelable promise. By using a user-defined type guard that makes TS understand it's saying something is a specific type

```suggestion
  static isCancelablePromise(promise: unknown): promise is CancelablePromise<unknown> {
```

```
const a: any = someUnknownThing;
if (isCancelablePromise(a)) {
  // TS will understand a is now type CancelablePromise<unknown> in this block
}
```